runtime/integration-tests/src/generic/cases/liquidity_pools.rs

```diff
@@ -26,126 +75,7098 @@ use xcm::{
+					<T as frame_system::Config>::RuntimeOrigin::root(),
+					MOONBEAM_EVM_CHAIN_ID,
+					[7; 20]
+				));
+			});
+		}
+
+		fn update_tranche_token_metadata<T: Runtime + FudgeSupport>() {
+			let mut env = FudgeEnv::<T>::from_parachain_storage(
+				Genesis::default()
+					.add(genesis::balances::<T>(cfg(1_000)))
+					.add(genesis::tokens::<T>(vec![(
+						GLMR_CURRENCY_ID,
+						DEFAULT_BALANCE_GLMR,
+					)]))
+					.storage(),
+			);
+
+			setup_test(&mut env);
+
+			env.parachain_state_mut(|| {
+				let pool_id = DEFAULT_POOL_ID;
+				// NOTE: Default pool admin is BOB
+				create_ausd_pool::<T>(pool_id);
+
+				// Missing tranche token should throw
+				let nonexistent_tranche = [71u8; 16];
+
+				assert_noop!(
+					pallet_liquidity_pools::Pallet::<T>::update_tranche_token_metadata(
+						RawOrigin::Signed(Keyring::Alice.into()).into(),
+						pool_id,
+						nonexistent_tranche,
+						Domain::EVM(MOONBEAM_EVM_CHAIN_ID),
+					),
+					pallet_liquidity_pools::Error::<T>::TrancheNotFound
+				);
+
+				let tranche_id = default_tranche_id::<T>(pool_id);
+
+				// Should throw if called by anything but `PoolAdmin`
+				assert_noop!(
+					pallet_liquidity_pools::Pallet::<T>::update_tranche_token_metadata(
+						RawOrigin::Signed(Keyring::Alice.into()).into(),
+						pool_id,
+						tranche_id,
+						Domain::EVM(MOONBEAM_EVM_CHAIN_ID),
+					),
+					pallet_liquidity_pools::Error::<T>::NotPoolAdmin
+				);
+
+				assert_ok!(
+					pallet_liquidity_pools::Pallet::<T>::update_tranche_token_metadata(
+						RawOrigin::Signed(Keyring::Bob.into()).into(),
+						pool_id,
+						tranche_id,
+						Domain::EVM(MOONBEAM_EVM_CHAIN_ID),
+					)
+				);
+
+				// Edge case: Should throw if tranche exists but metadata does not exist
+				let tranche_currency_id = CurrencyId::Tranche(pool_id, tranche_id);
+
+				orml_asset_registry::Metadata::<T>::remove(tranche_currency_id);
+
+				assert_noop!(
+					pallet_liquidity_pools::Pallet::<T>::update_tranche_token_metadata(
+						RawOrigin::Signed(Keyring::Bob.into()).into(),
+						pool_id,
+						tranche_id,
+						Domain::EVM(MOONBEAM_EVM_CHAIN_ID),
+					),
+					pallet_liquidity_pools::Error::<T>::TrancheMetadataNotFound
+				);
+			});
+		}
+
+		crate::test_for_runtimes!([development], add_pool);
+		crate::test_for_runtimes!([development], add_tranche);
+		crate::test_for_runtimes!([development], update_member);
+		crate::test_for_runtimes!([development], update_token_price);
+		crate::test_for_runtimes!([development], add_currency);
+		crate::test_for_runtimes!([development], add_currency_should_fail);
+		crate::test_for_runtimes!([development], allow_investment_currency);
+		crate::test_for_runtimes!([development], allow_pool_should_fail);
+		crate::test_for_runtimes!([development], schedule_upgrade);
+		crate::test_for_runtimes!([development], cancel_upgrade);
+		crate::test_for_runtimes!([development], update_tranche_token_metadata);
```

Comment:
Out of curiosity, what's the current problem with the rest of the runtimes?

Reply:
I think it's mostly related to the native currency that we use there, and probably some other things that @NunoAlexandre might know better. However, I also think that later on we can probably improve these tests a bit more and try to run them for all runtimes, I'm sure there's a way to achieve that.